topicExtractor.py

```diff
@@ -152,14 +153,15 @@ def fitTopicModel(self):
                     logging.warning(
                         "All topics are -1. Retrying with K-means clustering..."
                     )
-                    # This import should not occur frequently, so we only call when it is needed.
-                    from sklearn.cluster import KMeans
+                    KMeansAttempt = self.useKMeans(vectorizerModel, docs)
 
-                    # We use 3 clusters as a default value. 
-                    # This means the transcript will be grouped into 3 topics.
-                    clusterModel = KMeans(n_clusters=3)
-                    self.initializeTopicModel(vectorizerModel, clusterModel)
-                    self.topics, probs = self.topicModel.fit_transform(docs)
+                    if not KMeansAttempt:
+                        logging.warn("Retrying without Vectorizer model.")
+                        KMeansAttempt = self.useKMeans(None, docs)
+
+                    if not KMeansAttempt:
```

Comment:
Can you share which video you are used to test this usecase/bug where you ended up writing this extra logic?

Reply:
This is the one video transcript ([Link to file on Drive](https://drive.google.com/drive/folders/1QOoTou7dDJGfA6amdw8mxKasUknZ44Cn?usp=drive_link)) that can cause the error.

Reply:
Thanks! I will run the PR in the afternoon.

Reply:
I ran the script in the 3 scenarios all cases generating the questions
1. Without KMean clustering
2. with KMean clustering
3. No Vector model

Reply:
That means that the script first tried HDBSCAN, failed, tried KMeans, failed, and then tried KMeans with no Vectorizer model and passed, in that order. That should be the expected sequence of events. I haven't found a case where it fails after all three attempts made.

Reply:
I used `Sakai, UDP Cohort and Credential` srt files for testing it. 

UDP Cohort works with HDBSCAN
Sakai works with KMean
Credential is the no vectormodel case.